crates/router/src/connector/paypal.rs

```diff
@@ -0,0 +1,705 @@
+mod transformers;
+use std::fmt::Debug;
+
+use base64::Engine;
+use error_stack::{IntoReport, ResultExt};
+use transformers as paypal;
+
+use self::transformers::PaypalMeta;
+use crate::{
+    configs::settings,
+    connector::utils::RefundsRequestData,
+    consts,
+    core::{
+        errors::{self, CustomResult},
+        payments,
+    },
+    headers,
+    services::{self, ConnectorIntegration, PaymentAction},
+    types::{
+        self,
+        api::{self, CompleteAuthorize, ConnectorCommon, ConnectorCommonExt},
+        ErrorResponse, Response,
+    },
+    utils::{self, BytesExt},
+};
+
+#[derive(Debug, Clone)]
+pub struct Paypal;
+
+impl api::Payment for Paypal {}
+impl api::PaymentSession for Paypal {}
+impl api::ConnectorAccessToken for Paypal {}
+impl api::PreVerify for Paypal {}
+impl api::PaymentAuthorize for Paypal {}
+impl api::PaymentsCompleteAuthorize for Paypal {}
+impl api::PaymentSync for Paypal {}
+impl api::PaymentCapture for Paypal {}
+impl api::PaymentVoid for Paypal {}
+impl api::Refund for Paypal {}
+impl api::RefundExecute for Paypal {}
+impl api::RefundSync for Paypal {}
+
+impl<Flow, Request, Response> ConnectorCommonExt<Flow, Request, Response> for Paypal
+where
+    Self: ConnectorIntegration<Flow, Request, Response>,
+{
+    fn build_headers(
+        &self,
+        req: &types::RouterData<Flow, Request, Response>,
+        _connectors: &settings::Connectors,
+    ) -> CustomResult<Vec<(String, String)>, errors::ConnectorError> {
+        let access_token = req
+            .access_token
+            .clone()
+            .ok_or(errors::ConnectorError::FailedToObtainAuthType)?;
+        let key = &req.payment_id;
```

Comment:
Pass attempt id, as during retries we will be using the same payment id but attempt id will be different

Reply:
done!